project-base/storefront/components/Blocks/Popup/ManageCustomerUserPopup.tsx

```diff
@@ -0,0 +1,248 @@
+import { FormColumn } from 'components/Forms/Lib/FormColumn';
+import { FormLine } from 'components/Forms/Lib/FormLine';
+import { FormLineError } from 'components/Forms/Lib/FormLineError';
+import { Select } from 'components/Forms/Select/Select';
+import { TextInputControlled } from 'components/Forms/TextInput/TextInputControlled';
+import { Popup } from 'components/Layout/Popup/Popup';
+import {
+    useCustomerUserManageProfileForm,
+    useCustomerUserManageProfileFormMeta,
+} from 'components/Pages/Customer/customerUserManageProfileFormMeta';
+import { useCurrentCustomerData } from 'connectors/customer/CurrentCustomer';
+import { useAddNewCustomerUserMutation } from 'graphql/requests/customer/mutations/AddNewCustomerUserMutation.generated';
+import { useEditCustomerUserPersonalDataMutation } from 'graphql/requests/customer/mutations/EditCustomerUserPersonalDataMutation.generated';
+import { TypeCompanyCustomerUser } from 'graphql/types';
+import { GtmMessageOriginType } from 'gtm/enums/GtmMessageOriginType';
+import useTranslation from 'next-translate/useTranslation';
+import { Controller, FormProvider, Path, SubmitHandler, UseFormReturn } from 'react-hook-form';
+import { useSessionStore } from 'store/useSessionStore';
+import { CustomerUserManageProfileFormType } from 'types/form';
+import { CombinedError } from 'urql';
+import { useCurrentCustomerUserPermissions } from 'utils/auth/useCurrentCustomerUserPermissions';
+import { getUserFriendlyErrors } from 'utils/errors/friendlyErrorMessageParser';
+import { useErrorPopup } from 'utils/forms/useErrorPopup';
+import { showErrorMessage } from 'utils/toasts/showErrorMessage';
+import { showSuccessMessage } from 'utils/toasts/showSuccessMessage';
+import { useCustomerUserGroupsAsSelectOptions } from 'utils/user/useCustomerUserGroupsAsSelectOptions';
+
+type ManageCustomerUserPopupProps = {
+    customerUser?: TypeCompanyCustomerUser;
+    mode?: 'edit' | 'add';
+};
+
+export const ManageCustomerUserPopup: FC<ManageCustomerUserPopupProps> = ({ customerUser, mode = 'edit' }) => {
+    const { t } = useTranslation();
+    const [, customerEditUser] = useEditCustomerUserPersonalDataMutation();
+    const [, customerAddUser] = useAddNewCustomerUserMutation();
+    const updatePortalContent = useSessionStore((s) => s.updatePortalContent);
+    const { canManageProfile } = useCurrentCustomerUserPermissions();
+    const customerUserRoleGroupsAsSelectOptions = useCustomerUserGroupsAsSelectOptions();
+    const currentCustomerUserData = useCurrentCustomerData(customerUser);
+
+    const customerUserDefaultFormData =
+        customerUser && currentCustomerUserData
+            ? {
+                  ...currentCustomerUserData,
+                  roleGroup: {
+                      label: currentCustomerUserData.roleGroup.name,
+                      value: currentCustomerUserData.roleGroup.uuid,
+                  },
+              }
+            : {
```

Comment:
I would create `const defaultValues = { ... }`

Reply:
Added connector to handle `customerUser` data separately from `currentCustomerData`